### ISSUE TITLE
Assertion fail under runTCO.

### ISSUE THREAD

**Issue by gmaxwell** [ElementsProject/simplicity#22]
Running this fuzzer generated test case with assertions enabled and test.c:testTx1 as the transaction context:

```
test: eval.c:383: runTCO: Assertion `0 == state.activeWriteFrame->offset' failed.

Program received signal SIGABRT, Aborted.
0x00007ffff7e17e35 in raise () from /lib64/libc.so.6
(gdb) bt
#0  0x00007ffff7e17e35 in raise () from /lib64/libc.so.6
#1  0x00007ffff7e02895 in abort () from /lib64/libc.so.6
#2  0x00007ffff7e02769 in __assert_fail_base.cold () from /lib64/libc.so.6
#3  0x00007ffff7e10566 in __assert_fail () from /lib64/libc.so.6
#4  0x0000000000416ca1 in runTCO (env=0x7fffffffd580, len=<optimized out>, type_dag=0x43d560, dag=0x43c570, stack=<optimized out>, state=...) at eval.c:383
#5  evalTCOExpression (evalSuccess=evalSuccess@entry=0x7fffffffd664, output=output@entry=0x0, outputSize=outputSize@entry=0, input=input@entry=0x0, 
    inputSize=inputSize@entry=0, dag=0x43c570, type_dag=0x43d560, len=<optimized out>, env=0x7fffffffd580) at eval.c:686
#6  0x000000000040ce2b in evalTCOProgram (env=0x7fffffffd580, len=14, type_dag=<optimized out>, dag=<optimized out>, evalSuccess=0x7fffffffd664)
    at primitive/../eval.h:47
#7  elements_simplicity_execSimplicity (success=success@entry=0x7fffffffd664, tx=<optimized out>, ix=ix@entry=0, cmr=cmr@entry=0x0, wmr=wmr@entry=0x0, 
    file=<optimized out>) at primitive/elements.c:491
```

The program is:

```
unsigned char t3out_24_crashes_id_000000_sig_06_src_006885_op_havoc_rep_2[] = {
  0xdd, 0x36, 0x6c, 0x00, 0x82, 0x50, 0x44, 0xab, 0x82, 0x51, 0x24, 0x03,
  0x8a, 0xd0, 0x21
};
unsigned int t3out_24_crashes_id_000000_sig_06_src_006885_op_havoc_rep_2_len = 15;
```

**Comment by roconnor-blockstream** [comment]
The program in question decodes as the (untyped) simplicity dag:

    [Prim (someArrow outputNullDatum)
    ,Prim (someArrow outputNullDatum)
    ,Prim (someArrow version)
    ,Pair () () () 1 1
    ,Injl () () () 3
    ,Comp () () () 2 1
    ,Prim (someArrow outputAsset)
    ,Prim (someArrow outputsHash)
    ,Pair () () () 1 3
    ,Pair () () () 1 2
    ,Prim (someArrow version)
    ,Witness () () ()
    ,Pair () () () 3 3
    ,Comp () () () 1 2]

This will likely turn out to be the same underlying issue as https://github.com/ElementsProject/simplicity/issues/21.

**Comment by gmaxwell** [comment]
Fixed by #23
